File: iosApp/iosApp/Services/NodeServiceAssembler.swift
// Created by web3d3v on 18/10/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation
import web3walletcore

final class NodeServiceAssembler: AssemblerComponent {

    func register(to registry: AssemblerRegistry) {
        registry.register(scope: .instance) { resolver -> NodeService in
            DefaultNodeService()
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/NodeServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.services.node.DefaultNodeService
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope

class NodeServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register("NodeService", AssemblerRegistryScope.INSTANCE) {
            DefaultNodeService()
        }
    }
}